**Table S2**: Decomposed modules contributed from the different tools. The name of the module links to an active Galaxy instance where that can be used, and TS provides a second link to the module in the Galaxy Toolshed.
Each module is linked to one of the cli-layers mentioned in [Table S1](S1_cli-layer.md) and assigned to one or more of the relevant analysis areas: Clustering (**C**), Differential expression/Marker detection (**DE-MD**), Trajectories (**T**), Cell type alignment (**CT**) and Dimensionality reduction (**DR**).

| Module | Description | cli-layer | Analysis areas |
|--------|-------------|-----------|----------------|
| [anndata_ops](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fanndata_ops%2Fanndata_ops)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/anndata_ops) | modifies metadata and flags genes | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) |  |
| [scanpy_filter_cells](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_filter_cells%2Fscanpy_filter_cells)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_filter_cells) | based on counts and numbers of genes expressed | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | QC |
| [scanpy_filter_genes](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_filter_genes%2Fscanpy_filter_genes)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_filter_genes) | based on counts and numbers of cells expressed | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | QC |
| [scanpy_find_cluster](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_find_cluster%2Fscanpy_find_cluster)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_find_cluster) | based on community detection on KNN graph | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | C |
| [scanpy_find_markers](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_find_markers%2Fscanpy_find_markers)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_find_markers) | to find differentially expressed genes between groups | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | DE-MD |
| [scanpy_find_variable_genes](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_find_variable_genes%2Fscanpy_find_variable_genes)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_find_variable_genes) | based on normalised dispersion of expression | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | P |
| [scanpy_integrate_bbknn](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_integrate_bbknn%2Fscanpy_integrate_bbknn)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_integrate_bbknn) | batch-balanced K-nearest neighbours | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) |  |
| [scanpy_integrate_combat](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_integrate_combat%2Fscanpy_integrate_combat)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_integrate_combat) | adjust expression for variables that might introduce batch effect | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) |  |
| [scanpy_integrate_harmony](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_integrate_harmony%2Fscanpy_integrate_harmony)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_integrate_harmony) | adjust principal components for variables that might introduce batch effect | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) |  |
| [scanpy_integrate_mnn](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_integrate_mnn%2Fscanpy_integrate_mnn)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_integrate_mnn) | correct batch effects by matching mutual nearest neighbors | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) |  |
| [scanpy_compute_graph](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_compute_graph%2Fscanpy_compute_graph)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_compute_graph) | to derive kNN graph | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | C T |
| [scanpy_normalise_data](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_normalise_data%2Fscanpy_normalise_data)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_normalise_data) | to make all cells having the same total expression | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | P |
| [scanpy_parameter_iterator](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_parameter_iterator%2Fscanpy_parameter_iterator)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_parameter_iterator) | produce an iteration over a defined parameter | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) |  |
| [scanpy_plot_embed](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_plot_embed%2Fscanpy_plot_embed)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_plot_embed) | visualise cell embeddings | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | C |
| [scanpy_plot_trajectory](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_plot_trajectory%2Fscanpy_plot_trajectory)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_plot_trajectory) | visualise cell trajectories | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | T |
| [scanpy_read_10x](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_read_10x%2Fscanpy_read_10x)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_read_10x) | into hdf5 object handled by scanpy | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) |  |
| [scanpy_regress_variable](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_regress_variable%2Fscanpy_regress_variable)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_regress_variable) | variables that might introduce batch effect | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | P |
| [scanpy_run_diffmap](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_diffmap%2Fscanpy_run_diffmap)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_diffmap) | calculate diffusion components | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | T |
| [scanpy_run_dpt](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_dpt%2Fscanpy_run_dpt)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_dpt) | diffusion pseudotime inference | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | T |
| [scanpy_run_fdg](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_fdg%2Fscanpy_run_fdg)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_fdg) | visualise cell clusters using force-directed graph | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) |  |
| [scanpy_run_paga](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_paga%2Fscanpy_run_paga)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_paga) | trajectory inference | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | T |
| [scanpy_run_pca](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_pca%2Fscanpy_run_pca)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_pca) | for dimensionality reduction | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | DR |
| [scanpy_run_tsne](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_tsne%2Fscanpy_run_tsne)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_tsne) | visualise cell clusters using tSNE | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | DR |
| [scanpy_run_umap](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_umap%2Fscanpy_run_umap)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_umap) | visualise cell clusters using UMAP | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | DR |
| [scanpy_scale_data](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_scale_data%2Fscanpy_scale_data)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_scale_data) | to make expression variance the same for all genes | [scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master) | P |
| [seurat_dim_plot](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_dim_plot%2Fseurat_dim_plot)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_dim_plot) | graphs the output of a dimensional reduction technique (PCA by default). Cells are colored by their identity class. | [r-seurat-scripts](https://github.com/ebi-gene-expression-group/r-seurat-scripts/tree/master) | DR |
| [seurat_export_cellbrowser](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_export_cellbrowser%2Fseurat_export_cellbrowser)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_export_cellbrowser) | produces files for UCSC CellBrowser import. | [r-seurat-scripts](https://github.com/ebi-gene-expression-group/r-seurat-scripts/tree/master) |  |
| [seurat_filter_cells](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_filter_cells%2Fseurat_filter_cells)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_filter_cells) | filter cells in a Seurat object | [r-seurat-scripts](https://github.com/ebi-gene-expression-group/r-seurat-scripts/tree/master) |  |
| [seurat_find_clusters](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_find_clusters%2Fseurat_find_clusters)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_find_clusters) | find clusters of cells | [r-seurat-scripts](https://github.com/ebi-gene-expression-group/r-seurat-scripts/tree/master) | C |
| [seurat_find_markers](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_find_markers%2Fseurat_find_markers)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_find_markers) | find markers (differentially expressed genes) | [r-seurat-scripts](https://github.com/ebi-gene-expression-group/r-seurat-scripts/tree/master) | DE-MD |
| [seurat_find_neighbours](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_find_neighbours%2Fseurat_find_neighbours)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_find_neighbours) | constructs a Shared Nearest Neighbor (SNN) Graph | [r-seurat-scripts](https://github.com/ebi-gene-expression-group/r-seurat-scripts/tree/master) | C |
| [seurat_find_variable_genes](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_find_variable_genes%2Fseurat_find_variable_genes)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_find_variable_genes) | identify variable genes | [r-seurat-scripts](https://github.com/ebi-gene-expression-group/r-seurat-scripts/tree/master) | C DE-MD |
| [seurat_normalise_data](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_normalise_data%2Fseurat_normalise_data)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_normalise_data) | normalise data | [r-seurat-scripts](https://github.com/ebi-gene-expression-group/r-seurat-scripts/tree/master) | C |
| [seurat_read10x](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_read10x%2Fseurat_read10x)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_read10x) | Loads Tabular or 10x data into a serialized seurat R object | [r-seurat-scripts](https://github.com/ebi-gene-expression-group/r-seurat-scripts/tree/master) |  |
| [seurat_run_pca](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_run_pca%2Fseurat_run_pca)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_run_pca) | run a PCA dimensionality reduction | [r-seurat-scripts](https://github.com/ebi-gene-expression-group/r-seurat-scripts/tree/master) | DR |
| [seurat_run_tsne](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_run_tsne%2Fseurat_run_tsne)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_run_tsne) | run t-SNE dimensionality reduction | [r-seurat-scripts](https://github.com/ebi-gene-expression-group/r-seurat-scripts/tree/master) | C DR |
| [seurat_scale_data](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_scale_data%2Fseurat_scale_data)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_scale_data) | scale and center genes | [r-seurat-scripts](https://github.com/ebi-gene-expression-group/r-seurat-scripts/tree/master) | C |
| [sc3_calc_biology](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_calc_biology%2Fsc3_calc_biology)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_calc_biology) | calculates DE genes, marker genes and cell outliers | [bioconductor-sc3-scripts](https://github.com/ebi-gene-expression-group/bioconductor-sc3-scripts/tree/master) | DE-MD |
| [sc3_calc_consens](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_calc_consens%2Fsc3_calc_consens)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_calc_consens) | from multiple runs of k-means clustering | [bioconductor-sc3-scripts](https://github.com/ebi-gene-expression-group/bioconductor-sc3-scripts/tree/master) | C |
| [sc3_calc_dists](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_calc_dists%2Fsc3_calc_dists)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_calc_dists) | between cells | [bioconductor-sc3-scripts](https://github.com/ebi-gene-expression-group/bioconductor-sc3-scripts/tree/master) |  |
| [sc3_calc_transfs](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_calc_transfs%2Fsc3_calc_transfs)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_calc_transfs) | of distances using PCA and graph Laplacian | [bioconductor-sc3-scripts](https://github.com/ebi-gene-expression-group/bioconductor-sc3-scripts/tree/master) | C |
| [sc3_estimate_k](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_estimate_k%2Fsc3_estimate_k)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_estimate_k) | the number of clusters for k-means clustering | [bioconductor-sc3-scripts](https://github.com/ebi-gene-expression-group/bioconductor-sc3-scripts/tree/master) | C |
| [sc3_kmeans](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_kmeans%2Fsc3_kmeans)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_kmeans) | perform k-means clustering | [bioconductor-sc3-scripts](https://github.com/ebi-gene-expression-group/bioconductor-sc3-scripts/tree/master) | C |
| [sc3_prepare](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_prepare%2Fsc3_prepare)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_prepare) | a sc3 SingleCellExperiment object | [bioconductor-sc3-scripts](https://github.com/ebi-gene-expression-group/bioconductor-sc3-scripts/tree/master) |  |
| [scmap_get_std_output](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscmap_get_std_output%2Fscmap_get_std_output)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scmap_get_std_output) | Create final output in standard format to allow for downstream analysis of predicted labels by tools of the EBI gene expression group's cell-types-analysis package | [scmap-cli](https://github.com/ebi-gene-expression-group/scmap-cli/tree/master) |  |
| [scmap_index_cell](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscmap_index_cell%2Fscmap_index_cell)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scmap_index_cell) | creates a cell index for a dataset to enable fast approximate nearest neighbour search | [scmap-cli](https://github.com/ebi-gene-expression-group/scmap-cli/tree/master) | CT |
| [scmap_index_cluster](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscmap_index_cluster%2Fscmap_index_cluster)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scmap_index_cluster) | calculates centroids of each cell type and merges them into a single table | [scmap-cli](https://github.com/ebi-gene-expression-group/scmap-cli/tree/master) | CT |
| [scmap_preprocess_sce](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscmap_preprocess_sce%2Fscmap_preprocess_sce)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scmap_preprocess_sce) | Pre-process SCE object before running downstream analysis | [scmap-cli](https://github.com/ebi-gene-expression-group/scmap-cli/tree/master) |  |
| [scmap_scmap_cell](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscmap_scmap_cell%2Fscmap_scmap_cell)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scmap_scmap_cell) | searches each cell in a query dataset for the nearest neighbours by cosine distance within a collection of reference datasets. | [scmap-cli](https://github.com/ebi-gene-expression-group/scmap-cli/tree/master) | CT |
| [scmap_scmap_cluster](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscmap_scmap_cluster%2Fscmap_scmap_cluster)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scmap_scmap_cluster) | projects one dataset to another | [scmap-cli](https://github.com/ebi-gene-expression-group/scmap-cli/tree/master) | CT |
| [scmap_select_features](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscmap_select_features%2Fscmap_select_features)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scmap_select_features) | finds the most informative features (genes/transcripts) for projection. | [scmap-cli](https://github.com/ebi-gene-expression-group/scmap-cli/tree/master) | CT |
| [scater_calculate_cpm](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscater_calculate_cpm%2Fscater_calculate_cpm)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scater_calculate_cpm) | from raw counts | [bioconductor-scater-scripts](https://github.com/ebi-gene-expression-group/bioconductor-scater-scripts/tree/master) |  |
| [scater_calculate_qc_metrics](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscater_calculate_qc_metrics%2Fscater_calculate_qc_metrics)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scater_calculate_qc_metrics) | based on expression values and experiment information | [bioconductor-scater-scripts](https://github.com/ebi-gene-expression-group/bioconductor-scater-scripts/tree/master) |  |
| [scater_filter](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscater_filter%2Fscater_filter)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scater_filter) | cells and genes based on pre-calculated stats and QC metrics | [bioconductor-scater-scripts](https://github.com/ebi-gene-expression-group/bioconductor-scater-scripts/tree/master) |  |
| [scater_is_outlier](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscater_is_outlier%2Fscater_is_outlier)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scater_is_outlier) | cells based on expression metrics | [bioconductor-scater-scripts](https://github.com/ebi-gene-expression-group/bioconductor-scater-scripts/tree/master) |  |
| [scater_normalize](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscater_normalize%2Fscater_normalize)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scater_normalize) | expression values by library size in log2 scale | [bioconductor-scater-scripts](https://github.com/ebi-gene-expression-group/bioconductor-scater-scripts/tree/master) | C |
| [scater_read_10x_results](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscater_read_10x_results%2Fscater_read_10x_results)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scater_read_10x_results) | Loads 10x data into a serialized scater R object | [bioconductor-scater-scripts](https://github.com/ebi-gene-expression-group/bioconductor-scater-scripts/tree/master) |  |
| [monocle3_create](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_create%2Fmonocle3_create)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_create) | a Monocle3 object from input data | [monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master) |  |
| [monocle3_diffExp](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_diffexp%2Fmonocle3_diffExp)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_diffexp) |  of genes along a trajectory | [monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master) | DE-MD |
| [monocle3_learnGraph](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_learngraph%2Fmonocle3_learnGraph)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_learngraph) | between cells in dimensionality reduced space | [monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master) | T |
| [monocle3_orderCells](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_ordercells%2Fmonocle3_orderCells)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_ordercells) | along trajectories | [monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master) | T |
| [monocle3_partition](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_partition%2Fmonocle3_partition)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_partition) | of cells into groups | [monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master) | C T |
| [monocle3_plotCells](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_plotcells%2Fmonocle3_plotCells)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_plotcells) |  in the reduced dimensionality space | [monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master) | T |
| [monocle3_preprocess](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_preprocess%2Fmonocle3_preprocess)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_preprocess) | a Monocle3 object to an initially dimensionally reduced space | [monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master) | P |
| [monocle3_reduceDim](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_reducedim%2Fmonocle3_reduceDim)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_reducedim) | for downstream analysis | [monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master) | DR |
| [monocle3_topmarkers](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_topmarkers%2Fmonocle3_topmarkers)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_topmarkers) | for cell groups | [monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master) |  |
| [run_sccaf](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Frun_sccaf%2Frun_sccaf)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/run_sccaf) | to assess and optimise clustering | None | C |
| [sccaf_asses](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsccaf_asses%2Fsccaf_asses)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sccaf_asses) | runs an assesment of an SCCAF optimisation result or an existing clustering. | None |  |
| [sccaf_asses_merger](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsccaf_asses_merger%2Fsccaf_asses_merger)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sccaf_asses_merger) | brings together distributed assesments. | None | C |
| [sccaf_regress_out](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsccaf_regress_out%2Fsccaf_regress_out)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sccaf_regress_out) | with multiple categorical keys on an AnnData object. | None |  |
| [sceasy_convert](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsceasy_convert%2Fsceasy_convert)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sceasy_convert) | a data object between formats | None | C |
| [ucsc_cell_browser](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fucsc_cell_browser%2Fucsc_cell_browser)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/ucsc_cell_browser) | displays single-cell clusterized data in an interactive web application. | None |  |
| [hca_matrix_downloader](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fhca_matrix_downloader%2Fhca_matrix_downloader)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/hca_matrix_downloader) | retrieves expression matrices and metadata from the Human Cell Atlas. | [hca-matrix-downloader](https://github.com/ebi-gene-expression-group/hca-matrix-downloader/tree/master) |  |
| [retrieve_scxa](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fretrieve_scxa%2Fretrieve_scxa)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/retrieve_scxa) | Retrieves expression matrixes and metadata from EBI Single Cell Expression Atlas (SCXA) | None |  |
| [dropletutils_empty_drops](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fdropletutils_empty_drops%2Fdropletutils_empty_drops)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/dropletutils_empty_drops) | Distinguish between droplets containing cells and ambient RNA in a droplet-based single-cell RNA sequencing experiment. | [dropletutils-scripts](https://github.com/ebi-gene-expression-group/dropletutils-scripts/tree/master) |  |
| [dropletutils_read_10x](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fdropletutils_read_10x%2Fdropletutils_read_10x)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/dropletutils_read_10x) | into SingleCellExperiment object | [dropletutils-scripts](https://github.com/ebi-gene-expression-group/dropletutils-scripts/tree/master) |  |
| [garnett_check_markers](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fgarnett_check_markers%2Fgarnett_check_markers)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/garnett_check_markers) | Check marker file to filter out markers of suboptimal quality | [garnett-cli](https://github.com/ebi-gene-expression-group/garnett-cli/tree/master) |  |
| [garnett_classify_cells](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fgarnett_classify_cells%2Fgarnett_classify_cells)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/garnett_classify_cells) | Classify cells into cell types | [garnett-cli](https://github.com/ebi-gene-expression-group/garnett-cli/tree/master) |  |
| [garnett_get_feature_genes](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fgarnett_get_feature_genes%2Fgarnett_get_feature_genes)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/garnett_get_feature_genes) | Obtain a list of genes used as features in classification model | [garnett-cli](https://github.com/ebi-gene-expression-group/garnett-cli/tree/master) |  |
| [garnett_get_std_output](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fgarnett_get_std_output%2Fgarnett_get_std_output)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/garnett_get_std_output) | Get final output in standard format to allow for downstream analysis of predicted labels by tools of the EBI gene expression group's cell-types-analysis package | [garnett-cli](https://github.com/ebi-gene-expression-group/garnett-cli/tree/master) |  |
| [garnett_train_classifier](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fgarnett_train_classifier%2Fgarnett_train_classifier)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/garnett_train_classifier) | Train classifier based on marker gene list | [garnett-cli](https://github.com/ebi-gene-expression-group/garnett-cli/tree/master) |  |
| [garnett_transform_markers](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fgarnett_transform_markers%2Fgarnett_transform_markers)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/garnett_transform_markers) | Transform marker files from Single Cell Expression Atlas format to that compatible with Garnett | [garnett-cli](https://github.com/ebi-gene-expression-group/garnett-cli/tree/master) |  |
| [update_marker_file](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fupdate_marker_file%2Fupdate_marker_file)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/update_marker_file) | Update marker file by filtering out suboptimal markers | [garnett-cli](https://github.com/ebi-gene-expression-group/garnett-cli/tree/master) |  |
| [scpred_eigen_decompose](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscpred_eigen_decompose%2Fscpred_eigen_decompose)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scpred_eigen_decompose) | Perform matrix eigen-decomposition; initialize object of scPred class | [scpred-cli](https://github.com/ebi-gene-expression-group/scpred-cli/tree/master) |  |
| [scpred_get_feature_space](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscpred_get_feature_space%2Fscpred_get_feature_space)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scpred_get_feature_space) | Get feature space for training matrix | [scpred-cli](https://github.com/ebi-gene-expression-group/scpred-cli/tree/master) |  |
| [scpred_get_std_output](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscpred_get_std_output%2Fscpred_get_std_output)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scpred_get_std_output) | This method allows to export predicted labels in a standardised format, simplifying downstream analyses. | [scpred-cli](https://github.com/ebi-gene-expression-group/scpred-cli/tree/master) |  |
| [scpred_predict_labels](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscpred_predict_labels%2Fscpred_predict_labels)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scpred_predict_labels) | Make cell type predictions using trained model. | [scpred-cli](https://github.com/ebi-gene-expression-group/scpred-cli/tree/master) |  |
| [scpred_preprocess_data](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscpred_preprocess_data%2Fscpred_preprocess_data)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scpred_preprocess_data) | Pre-process input dataset (no train/test split) | [scpred-cli](https://github.com/ebi-gene-expression-group/scpred-cli/tree/master) |  |
| [scpred_train_model](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscpred_train_model%2Fscpred_train_model)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scpred_train_model) | Train classification model | [scpred-cli](https://github.com/ebi-gene-expression-group/scpred-cli/tree/master) |  |
| [scpred_traint_test_split](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscpred_traint_test_split%2Fscpred_traint_test_split)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scpred_traint_test_split) | CPM normalise and partition into train/test data | [scpred-cli](https://github.com/ebi-gene-expression-group/scpred-cli/tree/master) |  |
| [ct_build_cell_ontology_dict](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fct_build_cell_ontology_dict%2Fct_build_cell_ontology_dict)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/ct_build_cell_ontology_dict) | Create a mapping from labels to CL terms | [cell-types-analysis](https://github.com/ebi-gene-expression-group/cell-types-analysis/tree/master) |  |
| [ct_combine_tool_outputs](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fct_combine_tool_outputs%2Fct_combine_tool_outputs)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/ct_combine_tool_outputs) | Combine predictions for single tool from multiple datasets | [cell-types-analysis](https://github.com/ebi-gene-expression-group/cell-types-analysis/tree/master) |  |
| [ct_get_consensus_outputs](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fct_get_consensus_outputs%2Fct_get_consensus_outputs)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/ct_get_consensus_outputs) | Get consensus outputs across multiple tools | [cell-types-analysis](https://github.com/ebi-gene-expression-group/cell-types-analysis/tree/master) |  |
| [ct_get_empirical_dist](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fct_get_empirical_dist%2Fct_get_empirical_dist)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/ct_get_empirical_dist) | Get empirical distribution for tool performance table | [cell-types-analysis](https://github.com/ebi-gene-expression-group/cell-types-analysis/tree/master) |  |
| [ct_get_tool_perf_table](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fct_get_tool_perf_table%2Fct_get_tool_perf_table)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/ct_get_tool_perf_table) | Get performance table for a list of outputs generated by various tools | [cell-types-analysis](https://github.com/ebi-gene-expression-group/cell-types-analysis/tree/master) |  |
| [ct_get_tool_pvals](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fct_get_tool_pvals%2Fct_get_tool_pvals)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/ct_get_tool_pvals) | Get p-values for tool performance metrics | [cell-types-analysis](https://github.com/ebi-gene-expression-group/cell-types-analysis/tree/master) |  |
