**Table O2**: Decomposed modules contributed from the different tools. The name of the module links to an active Galaxy instance where that can be used, and TS provides a second link to the module in the Galaxy Toolshed.
Each module is linked to one of the cli-layers mentioned in [Table O1](O1_cli-layer.md) and assigned to one or more of the relevant analysis areas: Clustering (**C**), Differential expression/Marker detection (**DE-MD**), Trajectories (**T**), Cell type alignment (**CT**) and Dimensionality reduction (**DR**).

| Module | Description | cli-layer | Analysis areas |
|--------|-------------|-----------|----------------|
| [scanpy_filter_cells](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_filter_cells%2Fscanpy_filter_cells)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_filter_cells) | based on counts and numbers of genes expressed | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) |  |
| [scanpy_filter_genes](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_filter_genes%2Fscanpy_filter_genes)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_filter_genes) | based on counts and numbers of cells expressed | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) |  |
| [scanpy_find_cluster](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_find_cluster%2Fscanpy_find_cluster)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_find_cluster) | based on community detection on KNN graph | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | C |
| [scanpy_find_markers](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_find_markers%2Fscanpy_find_markers)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_find_markers) | to find differentially expressed genes between groups | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | DE-MD |
| [scanpy_find_variable_genes](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_find_variable_genes%2Fscanpy_find_variable_genes)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_find_variable_genes) | based on normalised dispersion of expression | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | P |
| [scanpy_compute_graph](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_compute_graph%2Fscanpy_compute_graph)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_compute_graph) | to derive kNN graph | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | C T |
| [scanpy_normalise_data](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_normalise_data%2Fscanpy_normalise_data)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_normalise_data) | to make all cells having the same total expression | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | P |
| [scanpy_parameter_iterator](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_parameter_iterator%2Fscanpy_parameter_iterator)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_parameter_iterator) | produce an iteration over a defined parameter | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) |  |
| [scanpy_plot_embed](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_plot_embed%2Fscanpy_plot_embed)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_plot_embed) | visualise cell embeddings | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | C |
| [scanpy_plot_trajectory](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_plot_trajectory%2Fscanpy_plot_trajectory)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_plot_trajectory) | visualise cell trajectories | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | T |
| [scanpy_read_10x](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_read_10x%2Fscanpy_read_10x)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_read_10x) | into hdf5 object handled by scanpy | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) |  |
| [scanpy_regress_variable](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_regress_variable%2Fscanpy_regress_variable)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_regress_variable) | variables that might introduce batch effect | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | P |
| [scanpy_run_diffmap](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_diffmap%2Fscanpy_run_diffmap)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_diffmap) | calculate diffusion components | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | T |
| [scanpy_run_dpt](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_dpt%2Fscanpy_run_dpt)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_dpt) | diffusion pseudotime inference | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | T |
| [scanpy_run_fdg](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_fdg%2Fscanpy_run_fdg)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_fdg) | visualise cell clusters using force-directed graph | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) |  |
| [scanpy_run_paga](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_paga%2Fscanpy_run_paga)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_paga) | trajectory inference | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | T |
| [scanpy_run_pca](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_pca%2Fscanpy_run_pca)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_pca) | for dimensionality reduction | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | DR |
| [scanpy_run_tsne](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_tsne%2Fscanpy_run_tsne)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_tsne) | visualise cell clusters using tSNE | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | DR |
| [scanpy_run_umap](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_umap%2Fscanpy_run_umap)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_umap) | visualise cell clusters using UMAP | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | DR |
| [scanpy_scale_data](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_scale_data%2Fscanpy_scale_data)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_scale_data) | to make expression variance the same for all genes | [scanpy-scripts]([scanpy-scripts](https://github.com/ebi-gene-expression-group/scanpy-scripts/tree/master)) | P |
| [seurat_create_seurat_object](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_create_seurat_object%2Fseurat_create_seurat_object)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_create_seurat_object) | create a Seurat object | [seurat-scripts]([seurat-scripts](https://github.com/ebi-gene-expression-group/seurat-scripts/tree/master)) |  |
| [seurat_dim_plot](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_dim_plot%2Fseurat_dim_plot)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_dim_plot) | graphs the output of a dimensional reduction technique (PCA by default). Cells are colored by their identity class. | [seurat-scripts]([seurat-scripts](https://github.com/ebi-gene-expression-group/seurat-scripts/tree/master)) | DR |
| [seurat_export_cellbrowser](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_export_cellbrowser%2Fseurat_export_cellbrowser)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_export_cellbrowser) | produces files for UCSC CellBrowser import. | [seurat-scripts]([seurat-scripts](https://github.com/ebi-gene-expression-group/seurat-scripts/tree/master)) |  |
| [seurat_filter_cells](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_filter_cells%2Fseurat_filter_cells)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_filter_cells) | filter cells in a Seurat object | [seurat-scripts]([seurat-scripts](https://github.com/ebi-gene-expression-group/seurat-scripts/tree/master)) |  |
| [seurat_find_clusters](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_find_clusters%2Fseurat_find_clusters)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_find_clusters) | find clusters of cells | [seurat-scripts]([seurat-scripts](https://github.com/ebi-gene-expression-group/seurat-scripts/tree/master)) | C |
| [seurat_find_markers](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_find_markers%2Fseurat_find_markers)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_find_markers) | find markers (differentially expressed genes) | [seurat-scripts]([seurat-scripts](https://github.com/ebi-gene-expression-group/seurat-scripts/tree/master)) | DE-MD |
| [seurat_find_variable_genes](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_find_variable_genes%2Fseurat_find_variable_genes)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_find_variable_genes) | identify variable genes | [seurat-scripts]([seurat-scripts](https://github.com/ebi-gene-expression-group/seurat-scripts/tree/master)) | C DE-MD |
| [seurat_normalise_data](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_normalise_data%2Fseurat_normalise_data)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_normalise_data) | normalise data | [seurat-scripts]([seurat-scripts](https://github.com/ebi-gene-expression-group/seurat-scripts/tree/master)) | C |
| [seurat_read10x](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_read10x%2Fseurat_read10x)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_read10x) | Loads 10x data into a serialized seurat R object | [seurat-scripts]([seurat-scripts](https://github.com/ebi-gene-expression-group/seurat-scripts/tree/master)) |  |
| [seurat_run_pca](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_run_pca%2Fseurat_run_pca)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_run_pca) | run a PCA dimensionality reduction | [seurat-scripts]([seurat-scripts](https://github.com/ebi-gene-expression-group/seurat-scripts/tree/master)) | DR |
| [seurat_run_tsne](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_run_tsne%2Fseurat_run_tsne)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_run_tsne) | run t-SNE dimensionality reduction | [seurat-scripts]([seurat-scripts](https://github.com/ebi-gene-expression-group/seurat-scripts/tree/master)) | C DR |
| [seurat_scale_data](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_scale_data%2Fseurat_scale_data)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/seurat_scale_data) | scale and center genes | [seurat-scripts]([seurat-scripts](https://github.com/ebi-gene-expression-group/seurat-scripts/tree/master)) | C |
| [sc3_calc_biology](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_calc_biology%2Fsc3_calc_biology)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_calc_biology) | calculates DE genes, marker genes and cell outliers | [sc3-scripts]([sc3-scripts](https://github.com/ebi-gene-expression-group/sc3-scripts/tree/master)) | DE-MD |
| [sc3_calc_consens](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_calc_consens%2Fsc3_calc_consens)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_calc_consens) | from multiple runs of k-means clustering | [sc3-scripts]([sc3-scripts](https://github.com/ebi-gene-expression-group/sc3-scripts/tree/master)) | C |
| [sc3_calc_dists](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_calc_dists%2Fsc3_calc_dists)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_calc_dists) | between cells | [sc3-scripts]([sc3-scripts](https://github.com/ebi-gene-expression-group/sc3-scripts/tree/master)) |  |
| [sc3_calc_transfs](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_calc_transfs%2Fsc3_calc_transfs)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_calc_transfs) | of distances using PCA and graph Laplacian | [sc3-scripts]([sc3-scripts](https://github.com/ebi-gene-expression-group/sc3-scripts/tree/master)) | C |
| [sc3_estimate_k](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_estimate_k%2Fsc3_estimate_k)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_estimate_k) | the number of clusters for k-means clustering | [sc3-scripts]([sc3-scripts](https://github.com/ebi-gene-expression-group/sc3-scripts/tree/master)) | C |
| [sc3_kmeans](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_kmeans%2Fsc3_kmeans)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_kmeans) | perform k-means clustering | [sc3-scripts]([sc3-scripts](https://github.com/ebi-gene-expression-group/sc3-scripts/tree/master)) | C |
| [sc3_prepare](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fsc3_prepare%2Fsc3_prepare)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/sc3_prepare) | a sc3 SingleCellExperiment object | [sc3-scripts]([sc3-scripts](https://github.com/ebi-gene-expression-group/sc3-scripts/tree/master)) |  |
| [scmap_index_cell](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscmap_index_cell%2Fscmap_index_cell)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scmap_index_cell) | creates a cell index for a dataset to enable fast approximate nearest neighbour search | [scmap-cli]([scmap-cli](https://github.com/ebi-gene-expression-group/scmap-cli/tree/master)) | CT |
| [scmap_index_cluster](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscmap_index_cluster%2Fscmap_index_cluster)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scmap_index_cluster) | calculates centroids of each cell type and merges them into a single table | [scmap-cli]([scmap-cli](https://github.com/ebi-gene-expression-group/scmap-cli/tree/master)) | CT |
| [scmap_scmap_cell](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscmap_scmap_cell%2Fscmap_scmap_cell)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scmap_scmap_cell) | searches each cell in a query dataset for the nearest neighbours by cosine distance within a collection of reference datasets. | [scmap-cli]([scmap-cli](https://github.com/ebi-gene-expression-group/scmap-cli/tree/master)) | CT |
| [scmap_scmap_cluster](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscmap_scmap_cluster%2Fscmap_scmap_cluster)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scmap_scmap_cluster) | projects one dataset to another | [scmap-cli]([scmap-cli](https://github.com/ebi-gene-expression-group/scmap-cli/tree/master)) | CT |
| [scmap_select_features](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscmap_select_features%2Fscmap_select_features)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scmap_select_features) | finds the most informative features (genes/transcripts) for projection. | [scmap-cli]([scmap-cli](https://github.com/ebi-gene-expression-group/scmap-cli/tree/master)) | CT |
| [scater_calculate_cpm](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscater_calculate_cpm%2Fscater_calculate_cpm)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scater_calculate_cpm) | from raw counts | [scater-scripts]([scater-scripts](https://github.com/ebi-gene-expression-group/scater-scripts/tree/master)) |  |
| [scater_calculate_qc_metrics](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscater_calculate_qc_metrics%2Fscater_calculate_qc_metrics)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scater_calculate_qc_metrics) | based on expression values and experiment information | [scater-scripts]([scater-scripts](https://github.com/ebi-gene-expression-group/scater-scripts/tree/master)) |  |
| [scater_filter](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscater_filter%2Fscater_filter)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scater_filter) | cells and genes based on pre-calculated stats and QC metrics | [scater-scripts]([scater-scripts](https://github.com/ebi-gene-expression-group/scater-scripts/tree/master)) |  |
| [scater_is_outlier](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscater_is_outlier%2Fscater_is_outlier)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scater_is_outlier) | cells based on expression metrics | [scater-scripts]([scater-scripts](https://github.com/ebi-gene-expression-group/scater-scripts/tree/master)) |  |
| [scater_normalize](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscater_normalize%2Fscater_normalize)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scater_normalize) | expression values by library size in log2 scale | [scater-scripts]([scater-scripts](https://github.com/ebi-gene-expression-group/scater-scripts/tree/master)) | C |
| [scater_read_10x_results](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscater_read_10x_results%2Fscater_read_10x_results)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scater_read_10x_results) | Loads 10x data into a serialized scater R object | [scater-scripts]([scater-scripts](https://github.com/ebi-gene-expression-group/scater-scripts/tree/master)) |  |
| [monocle3_create](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_create%2Fmonocle3_create)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_create) | a Monocle3 object from input data | [monocle-scripts]([monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master)) |  |
| [monocle3_diffExp](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_diffExp%2Fmonocle3_diffExp)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_diffExp) |  of genes along a trajectory | [monocle-scripts]([monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master)) | DE-MD |
| [monocle3_learnGraph](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_learnGraph%2Fmonocle3_learnGraph)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_learnGraph) | between cells in dimensionality reduced space | [monocle-scripts]([monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master)) | T |
| [monocle3_orderCells](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_orderCells%2Fmonocle3_orderCells)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_orderCells) | along trajectories | [monocle-scripts]([monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master)) | T |
| [monocle3_partition](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_partition%2Fmonocle3_partition)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_partition) | of cells into groups | [monocle-scripts]([monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master)) | C T |
| [monocle3_plotCells](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_plotCells%2Fmonocle3_plotCells)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_plotCells) |  in the reduced dimensionality space | [monocle-scripts]([monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master)) | T |
| [monocle3_preprocess](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_preprocess%2Fmonocle3_preprocess)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_preprocess) | a Monocle3 object to an initially dimensionally reduced space | [monocle-scripts]([monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master)) | P |
| [monocle3_reduceDim](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fmonocle3_reduceDim%2Fmonocle3_reduceDim)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_reduceDim) | for downstream analysis | [monocle-scripts]([monocle-scripts](https://github.com/ebi-gene-expression-group/monocle-scripts/tree/master)) | DR |
| [ucsc_cell_browser](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fucsc_cell_browser%2Fucsc_cell_browser)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/ucsc_cell_browser) | displays single-cell clusterized data in an interactive web application. | [None]([None](https://github.com/ebi-gene-expression-group/None/tree/master)) |  |
| [hca_matrix_downloader](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fhca_matrix_downloader%2Fhca_matrix_downloader)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/hca_matrix_downloader) | retrieves expression matrices and metadata from the Human Cell Atlas. | [hca-matrix-downloader]([hca-matrix-downloader](https://github.com/ebi-gene-expression-group/hca-matrix-downloader/tree/master)) |  |
| [retrieve_scxa](https://humancellatlas.usegalaxy.eu/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fretrieve_scxa%2Fretrieve_scxa)<sup>[TS](https://toolshed.g2.bx.psu.edu/view/ebi-gxa/retrieve_scxa) | Retrieves expression matrixes and metadata from EBI Single Cell Expression Atlas (SCXA) | [None]([None](https://github.com/ebi-gene-expression-group/None/tree/master)) |  |

| Module | Description | cli-layer | Analysis areas |
|--------|-------------|-----------|----------------|

| Module | Description | cli-layer | Analysis areas |
|--------|-------------|-----------|----------------|
